packages/mobile/src/screens/edit-track-screen/screens/NFTCollectionsScreen.tsx

```diff
@@ -0,0 +1,242 @@
+import { useCallback, useMemo } from 'react'
+
+import type {
+  Nullable,
+  PremiumConditions,
+  EthCollectionMap,
+  SolCollectionMap,
+  CommonState,
+  Collectible
+} from '@audius/common'
+import { Chain, accountSelectors, collectiblesSelectors } from '@audius/common'
+import { useField } from 'formik'
+import { View, Image } from 'react-native'
+import { useSelector } from 'react-redux'
+
+import IconImage from 'app/assets/images/iconImage.svg'
+import { Text } from 'app/components/core'
+import { makeStyles, typography } from 'app/styles'
+
+import { ListSelectionScreen } from './ListSelectionScreen'
+
+const messages = {
+  collections: 'COLLECTIONS',
+  searchCollections: 'Search Collections'
+}
+
+const { getUserId } = accountSelectors
+const { getUserCollectibles, getSolCollections } = collectiblesSelectors
+
+const defaultCollectibles = { [Chain.Eth]: [], [Chain.Sol]: [] }
+
+const useStyles = makeStyles(({ spacing, palette, type }) => ({
+  item: {
+    flexDirection: 'row',
+    alignItems: 'center'
+  },
+  logo: {
+    marginRight: spacing(3),
+    borderWidth: 1,
+    borderColor: palette.neutralLight8,
+    borderRadius: spacing(1),
+    width: spacing(8),
+    height: spacing(8)
+  },
+  collectionName: {
+    fontFamily: typography.fontByWeight.demiBold,
+    fontSize: typography.fontSize.large,
+    width: spacing(50)
+  },
+  row: {
+    alignItems: 'center'
+  }
+}))
+
+export const NFTCollectionsScreen = () => {
+  const styles = useStyles()
+  const [{ value: premiumConditions }, , { setValue: setPremiumConditions }] =
+    useField<Nullable<PremiumConditions>>('premium_conditions')
+  const accountUserId = useSelector(getUserId)
+  const collectibles =
+    useSelector((state: CommonState) => {
+      if (!accountUserId) return defaultCollectibles
+      return getUserCollectibles(state, { id: accountUserId })
+    }) ?? defaultCollectibles
+  const solCollections = useSelector(getSolCollections)
+
+  // Ethereum collections
+  const ethCollectionMap: EthCollectionMap = useMemo(() => {
```

Comment:
main thought, is that if we need to memoize these, can we put them in the `useSelector`, that way they get memoized automatically cause it's only changed when data from redux changes

Reply:
makes sense. and done!